### DISCUSSION TITLE
Chat about future of LinkFree CLI

### DISCUSSION THREAD

**Discussion by Pradumnasaraf** [Pradumnasaraf/BioDrop-CLI#37] [Ideas]
This is an idea discussion to talk about the future of the [LinkFree CLI](https://github.com/Pradumnasaraf/LinkFree-CLI) 🚀. If you have something general to share, please use the general chat #41

**Comment by Ananya2001-an** [comment]
# Future enhancements

1. search filter for icons 
2. on no updates, message at the end should be "no updates made" rather than "updated successfully"
3. group and color optional keys for links

Hey @Pradumnasaraf I have created a list of all future tasks that we will have to do. If you want to add something you can do that too. 😊

**Reply by Pradumnasaraf** [reply]
Hey,

Thanks for putting it in one place. Sure I will add if something came to mind. Also, I think we should create a separate issue for each so that we can welcome more contributors otherwise no one shows us we can fix them.

**Reply by Pradumnasaraf** [reply]
>  check if url entered by user is valid or not

I think we should not implement this because it can be too much gatekeeping (This is why we don't do this on LinkFree too). 

What I mean to say is that we don't wanna create too many hurdles for people in creating their profiles. Also, they can come again and update their profile. So extra contributions from them

Hope you got my point of view.

**Reply by Ananya2001-an** [reply]
> >  check if url entered by user is valid or not
> 
> I think we should not implement this because it can be too much gatekeeping (This is why we don't do this on LinkFree too). 
> 
> What I mean to say that we don't wanna create too many hurdles for people to increase their profile. Also, they can come again and update their profile. So extra contributions from them

Sure got it.

**Reply by Ananya2001-an** [reply]
> Hey,
> 
> Thanks for putting it in one place. Sure I will add if something came to mind. Also, I think we should create a separate issue for each so that we can welcome more contributors otherwise no one shows us we can fix them.

Sound good.

**Reply by Pradumnasaraf** [reply]
Awesome. Would you like to create issues?

**Reply by Ananya2001-an** [reply]
> Awesome. Would you like to create issues?

Yeah no problem

**Reply by Pradumnasaraf** [reply]
Nice. Also, I have converted this to `idea chat` because we started talking about features, so I created a new general #41, where people can ask for help. Don't wanna mix conversations.

**Reply by Ananya2001-an** [reply]
> Nice. Also, I have converted this to `idea chat` because we started talking about features, so I created a new general #41, where people can ask for help. Don't wanna mix conversations.

Cool

**Reply by Pradumnasaraf** [reply]
Hey, @Ananya2001-an 

In the issue, we should never put `help wanted`. It's used when we are seeking help in an emergency. I know you didn't know about this, so no worries. I have removed it. Thanks for opening the issue. :).

**Reply by Pradumnasaraf** [reply]
Also, do you think #43 is a `good first issue`? Seeing we have icons implemented in so many places.

Edit: same for the #44

**Reply by Ananya2001-an** [reply]
> Also, do you think #43 is a `good first issue`? Seeing we have icons implemented in so many places.

Yeah I guess not. But since we have to write code for it once and then copy everywhere else so maybe that won't be that difficult don't you think?

**Reply by Ananya2001-an** [reply]
#44 was not a good first issue I have removed that label.

**Reply by Pradumnasaraf** [reply]
> Yeah I guess not. But since we have to write code for it once and then copy everywhere else so maybe that won't be that difficult don't you think?

Yes it's 1000% not, and we still haven't written the code, so we can't assume things :). Use `good first issue` wisely, it's good and bad as well. Use it in the issue which is a one-liner code chnage and simple to fix. 

Like the color thing I fixed yesterday.

**Reply by Ananya2001-an** [reply]
> > Yeah I guess not. But since we have to write code for it once and then copy everywhere else so maybe that won't be that difficult don't you think?
> 
> Yes it's 1000% not, and we still haven't written the code, so we can't assume things :). Use `good first issue` wisely, it's good and bad as well. Use it in the issue which is a one-liner code chnage and simple to fix.

Okay got it. I will keep this in mind for the future. Thanks for the advice 😊

**Reply by Pradumnasaraf** [reply]
No worries. We all learning. I have removed that. Thank you for understanding.

**Reply by Pradumnasaraf** [reply]
Hey @Ananya2001-an ,

I have invited you to the LInkFree CLI project board with write access. The board help us a little bit to keep track of the issue and other things related to that

**Reply by Pradumnasaraf** [reply]
LinkFree CLI docs is live - https://linkfree.eddiehub.io/docs. I am happy and my work is done 🎉

**Reply by Ananya2001-an** [reply]
> LinkFree CLI docs is live - https://linkfree.eddiehub.io/docs. I am happy and my work is done 🎉 

Awesome 💖😊

**Reply by Ananya2001-an** [reply]
Hey @Pradumnasaraf I think we should add a sample data folder for people to test things locally or at least add info about it in the Readme that you have to create a data folder in the root directory so as to test things properly.

**Reply by Ananya2001-an** [reply]
This should be in the Contributing.md file

**Reply by Pradumnasaraf** [reply]
Hey, 

Great point, Ananya. I think it will be good to add in README. I will create a section in that. If we add this in contributing many people will miss it. What do you say?

**Reply by Pradumnasaraf** [reply]
Also, I have already PR #47 opened, it's good you didn't merge it. I will include thatchnages in the PR.

**Reply by Ananya2001-an** [reply]
> Hey, 
> 
> Great point, Ananya. I think it will be good to add in README it will create a section in that. If we add this in contributing many people will miss it. What do you say?

Yes that will be good. You can add that in the Readme itself.

**Reply by Pradumnasaraf** [reply]
Hey, @Ananya2001-an 

Just in case you're waiting for my review 1st on the PRs, you don't have to, feel free to review 1st and give feedback :)

**Reply by Pradumnasaraf** [reply]
You merged with the wrong convention by the way :)

**Reply by Ananya2001-an** [reply]
> You merged with the wrong convention by the way :)

I thought it was squash and merge wasn't it?

**Reply by Pradumnasaraf** [reply]
Nope the commit convention

**Reply by Ananya2001-an** [reply]
Ahh okay yeah I should have added the fix: prefix. My bad I didn't notice it.

**Reply by Pradumnasaraf** [reply]
<img width="308" alt="Screenshot 2023-02-26 at 11 15 38 AM" src="https://user-images.githubusercontent.com/51878265/221394492-c04fd762-7fef-465c-8a95-deadf77a3ed6.png">

We need to be careful while merging because it triggers the npm release. on `fix:` and `feat:`

**Reply by Ananya2001-an** [reply]
> <img alt="Screenshot 2023-02-26 at 11 15 38 AM" width="308" src="https://user-images.githubusercontent.com/51878265/221394492-c04fd762-7fef-465c-8a95-deadf77a3ed6.png">
> 
> We need to be careful while merging because it triggers the npm release. on `fix:` and `feat:`

Yeah I know. Really sorry for the silly mistake. I will be extra careful next time.

**Reply by Pradumnasaraf** [reply]
No worries. It happens.

**Reply by Pradumnasaraf** [reply]
Also, I think we should wait for two approvals (from you and me) and then Merge PR because there might be something that a single person can't see

For eg: 
<img width="379" alt="Screenshot 2023-02-26 at 11 26 01 AM" src="https://user-images.githubusercontent.com/51878265/221394854-6d7f7b9c-1b52-4c7e-be5a-cc7341636370.png">

I was dropping the review for that, but you merge till then 😆 . I also did this kind of mistake at the very start. So nothing to worry about

**Reply by Ananya2001-an** [reply]
> Also, I think we should wait for two approvals (from you and me) and then Merge PR because there might be something that a single person can't see
> 
> For eg: <img alt="Screenshot 2023-02-26 at 11 26 01 AM" width="379" src="https://user-images.githubusercontent.com/51878265/221394854-6d7f7b9c-1b52-4c7e-be5a-cc7341636370.png">
> 
> I was dropping the review for that, but you merge till then 😆 . I also did this kind of mistake at the very start. So nothing to worry about

Yeah I also think that waiting for both of our reviews is much better but I thought maybe you might be busy right now and the changes were not that big so I merged it. 😂 Will definitely follow the 2 reviews rule from now on :)

**Reply by Pradumnasaraf** [reply]
Haha. No worries. It's not a rule. Just to get different perspectives.

Also, did you thought that the fix will be that simple #42? I thought we need to do some kind of key check or json check at the end. But this was simple, quick, and clever.

**Reply by Ananya2001-an** [reply]
> Haha. No worries. It's not a rule. Just to get different perspectives.
> 
> Also, did you thought that the fix will be that simple #42? I thought we need to do some kind of key check or json check at the end. But this was simple, quick, and clever.

Yeah actually the fix was really simple. He did a good job. I guess we were overcomplicating things in our head but that's the usefulness of having other people to review your code since they think with a fresh perspective. You can't do all things on your own sometimes. Better to leave it for others ;)

**Reply by Pradumnasaraf** [reply]
> Yeah actually the fix was really simple. He did a good job. I guess we were overcomplicating things in our head but that's the usefulness of having other people to review your code since they think with a fresh perspective. You can't do all things on your own sometimes. Better to leave it for others ;)

AGREE 100%. This is why I said we should raise the issues and let the community contribute instead of fixing ourselves. Sometimes it blows my mind.

**Reply by Ananya2001-an** [reply]
> > Yeah actually the fix was really simple. He did a good job. I guess we were overcomplicating things in our head but that's the usefulness of having other people to review your code since they think with a fresh perspective. You can't do all things on your own sometimes. Better to leave it for others ;)
> 
> AGREE 100%. This is why I said we should raise the issues and let the community contribute. Sometimes it blows my mind.

True

**Reply by Pradumnasaraf** [reply]
Hey @Ananya2001-an 

I am planning to store `icon.json` locally so we don't have to do the fetch call. Also, change the structure of the JSON because we don't need a key value we can use an array.  Any hint why I am getting this length error from the enquirer(I guess)

<img width="1493" alt="Screenshot 2023-02-26 at 4 38 22 PM" src="https://user-images.githubusercontent.com/51878265/221406952-dc80ed32-cb3e-4145-a2ac-55e45b40b6f1.png">

I have pushed changes to a branch you can have a look also 
https://github.com/Pradumnasaraf/LinkFree-CLI/tree/fix-icon

**Reply by Ananya2001-an** [reply]
> Hey @Ananya2001-an
> 
> I am planning to store `icon.json` locally so we don't have to do the fetch call. Also, change the structure of the JSON because we don't need a key value we can use an array. Any hint why I am getting this length error from the enquirer(I guess)
> 
> <img alt="Screenshot 2023-02-26 at 4 38 22 PM" width="1493" src="https://user-images.githubusercontent.com/51878265/221406952-dc80ed32-cb3e-4145-a2ac-55e45b40b6f1.png">
> 
> I have pushed changes to a branch you can have a look also https://github.com/Pradumnasaraf/LinkFree-CLI/tree/fix-icon

I was also thinking the same. Sure I will look into the changes and let u know.

**Reply by Pradumnasaraf** [reply]
Oh, nice. Thank you. 
I am stuck for 2 hrs 😅 . both are returning objects not sure what is happing.

**Reply by Ananya2001-an** [reply]
> Oh, nice. Thank you. I am stuck for 2 hrs 😅 . both are returning objects not sure what is happing.

No worries I will inspect.

**Reply by Pradumnasaraf** [reply]
Hey @Ananya2001-an, If you trying to fix it then you can stop it. 

I asked one of the maintainers to give me the JSON file of all the icons supported in LInkFree (because the current file has ~60-70 icons and linkfree supports ~3500). They said they are getting icons in JSON format from `react-icons` package and they said you can directly implement that as well. I am trying that out and I think it will work.

**Reply by Pradumnasaraf** [reply]
Also, after that, we need to implement the search icon ASAP because it will show 3500+ icons than in the prompt 😆 .

**Reply by Ananya2001-an** [reply]
> Hey @Ananya2001-an, If you trying to fix it then you can stop it. 
> 
> I asked one of the maintainers to give me the JSON file of all the icons supported in LInkFree (because the current file has ~60-70 icons and linkfree supports ~3500). They said they are getting icons in JSON format from `react-icons` package and they said you can directly implement that as well. I am trying that out and I think it will work.

Okay cool. I was trying from my end and I was also getting the undefined error for some reason.

**Reply by Pradumnasaraf** [reply]
> Okay cool. I was trying from my end and I was also getting the undefined error for some reason.

Yes, that is so weird. You can leave that.

**Reply by Ananya2001-an** [reply]
> Also, after that, we need to implement the search icon ASAP because it will show 3500+ icons than in the prompt 😆 .

Yeah that it would be horrifying without it.

**Reply by Pradumnasaraf** [reply]
Hey, it worked 🎉 

Wow, I started working to fix the block host and end up implementing all the icons without a fetch call. You never know, haha

https://user-images.githubusercontent.com/51878265/221420276-c8ce6711-b771-40cd-b189-7c99f34ba857.mov

**I have created a PR as well. You can test it.** https://github.com/Pradumnasaraf/LinkFree-CLI/pull/50

**Reply by Ananya2001-an** [reply]
> Hey, it worked 🎉
> 
> Wow, I started working to fix the block host and end up implementing all the icons without a fetch call. You never know, haha
> 
> **I have created a PR as well. You can test it.** #50

Cool I will review it.

**Reply by Pradumnasaraf** [reply]
Awesome. We need not worry too much about 3500+ icons because the LinkFree still runs `v2.0.1` changes. But hope we will soon get #43 merged.

Wow, I thought to relax on Sunday a little bit, but it went so good 🤣

**Reply by Ananya2001-an** [reply]
> Awesome. We need not worry too much about 3500+ icons because the LinkFree still runs `v2.0.1` changes. But hope we will soon get #43 merged.
> 
> Wow, I thought to relax on Sunday a little bit, but it went so good 🤣

Yeah that would be good. My sundays are always like weekdays. TBH I can't live without working a whole day. I will get into panic mode real quick 😂

**Comment by Ananya2001-an** [comment]
Hey @Pradumnasaraf I was just thinking that why don't we move the CLI project to the EddieHubCommunity? I mean ultimately it's for LinkFree hence that way we can also get more contributors don't you think?

**Reply by Pradumnasaraf** [reply]
Hey, @Ananya2001-an 

Yes, you are 100% correct. This is the reason (for getting more contributors and traffic)I was not planning to do that because I am already overwhelmed by the notifications from my repos as well as community repos. And putting this Org means I have to lead and handle tons of user activities.

**Reply by Ananya2001-an** [reply]
> Hey, @Ananya2001-an 
> 
> Yes, you are 100% correct. This is the reason (for getting more contributors and traffic)I was not planning to do that because I am already overwhelmed by the notifications from my repos as well as community repos. And putting this Org means I have to lead and handle tons of user activities.

Gotcha

**Comment by Ananya2001-an** [comment]
Hey @Pradumnasaraf I had noticed this thing before but forgot to mention. There is one thing we would need to fix later. When we update any prop and we again choose to update, it doesn't show us the latest updates that were made. It might confuse the users as to whether their changes were implemented or not.

![image](https://user-images.githubusercontent.com/55504616/225216518-840f57c3-0585-4e5e-9630-7e95378d12aa.png)

**Reply by Pradumnasaraf** [reply]
Great point, @Ananya2001-an. We will do this after the current issues are done.

**Reply by Ananya2001-an** [reply]
Sure

**Comment by Pradumnasaraf** [comment]
Hey, @Ananya2001-an. Finally, #43 got merged wooooh!

There are a couple of minor things that need to fix in the icon thing, like the colour when we select from the list. It's not very accessible. I will raise a couple of PR in a few hr. Don't want to block #43 PR also it will take time to ask them to that

Also, then we can update the version In LinkFree

**Reply by Ananya2001-an** [reply]
Sure that sounds good to me. It was nice of Pratik to stay consistent on it. 😃 Ask me for any help if you want.

**Reply by Pradumnasaraf** [reply]
I agree. I hope they learned something as I do. 

Sure, I will, Thank you.

**Reply by Pradumnasaraf** [reply]
Hey, @Ananya2001-an. 

I think we should update the version in the LinkFree. Before that, do you wan't to do a deep test?

**Reply by Ananya2001-an** [reply]
I think its okay. I am pretty sure the changes are right because we checked at every commit. And I know you must have also checked it thoroughly so it would be fine IMO. You can bump the version.

**Reply by Pradumnasaraf** [reply]
Thank you. 

One thing is missing. I think we should mention something like in bracket **(Start typing to search icon)** otherwise, people will think we have only 10 options to select from the drop-down and we can't type. WDYT?

<img width="357" alt="Screenshot 2023-03-18 at 8 54 24 PM" src="https://user-images.githubusercontent.com/51878265/226115114-0bc9adbc-bd09-4a73-b51c-1d2453aaa42e.png">

**Reply by Ananya2001-an** [reply]
Yeah that will be good. I thought we already had that. Maybe I didn't check.

**Reply by Pradumnasaraf** [reply]
Would you want to do it?

**Reply by Pradumnasaraf** [reply]
Also, it's sad I didn't see it earlier. Now we have do it as a `fix:` and it's not a fix (so that it gets released on NPM 🤦 )

**Reply by Ananya2001-an** [reply]
I would love to. But I am doing some other work. Since it's a small change I think u can do it. I am sorry if I am putting everything on u to do.

**Reply by Ananya2001-an** [reply]
> Also, it's sad I didn't see it earlier. Now we have do it as a `fix:` and it's not a fix (so that it gets released on NPM 🤦 )

Oh that sucks..

**Reply by Pradumnasaraf** [reply]
> I would love to. But I am doing some other work. Since it's a small change I think u can do it. I am sorry if I am putting everything on u to do.

No, you are not putting. We have other works to do as well. We have to understand that. Even thank you for replying and reviewing PR even tho you are busy

**Reply by Pradumnasaraf** [reply]
I will go ahaed and great a PR now.

**Reply by Ananya2001-an** [reply]
Well I am not THAT kind of busy that I can't even give a review 😉  Thanks for being understanding.

**Reply by Pradumnasaraf** [reply]
Thank you.

Also, I have raised the PR. Feel free to take a look when you are free.

**Reply by Ananya2001-an** [reply]
Sure I will.

**Reply by Pradumnasaraf** [reply]
Merged https://github.com/EddieHubCommunity/LinkFree/pull/5541

**Reply by Ananya2001-an** [reply]
> Merged https://github.com/EddieHubCommunity/LinkFree/pull/5541

Great!
